doc/source/developers_guide.rst

```diff
@@ -41,6 +41,7 @@ Requirements
     * Python_ 3.8 or later
     * numpy_ >= 1.19.5
     * quantities_ >= 0.14.1
+    * joblib >= 1.2.0
```

Comment:
```suggestion
    * joblib >= 1.0.0
```

Reply:
I have not tested `joblib<1.2.0`. Looking at [joblib's changelog](https://github.com/joblib/joblib/blob/master/CHANGES.rst) I'm not sure if the bugs fixed since 1.0.0 affect the caching in neo, but I'd guess no and hope that a test will show if it does.

Speaking of tests, as the missing dependency did not pop up before, is there actually any test for the caching?